src/api/tests/sds_data_tests/test_sds_data.py

```diff
@@ -0,0 +1,259 @@
+    current_dir = os.path.dirname(__file__)
+    file_path = os.path.join(current_dir, "data", filename)
+    with open(file_path, "r") as file:
+        data = json_load(file)
+
+    transformed_data = []
+    for item in data:
+        transformed_item = {"params": {}, "path": item.get("path")}
+
+        for key, value in item.items():
+            if ".params." in key:
+                # Extract the actual param name by splitting the key
+                param_name = key.split(".params.")[-1]
+                transformed_item["params"][param_name] = urllib.parse.unquote(value)
+            elif key == "path":
+                # Path key remains as is
+                transformed_item["path"] = value
+        transformed_data.append((transformed_item))
+    test_items = random.sample(transformed_data, test_count)
+    return test_items
+
+
+def _assert_response_match(ldap_body_reordered, cpm_body_reordered, item):
+    assert len(cpm_body_reordered["entry"]) == len(
+        ldap_body_reordered["entry"]
+    ), f"Entries do not match the total. When calling with {item}"
+    for idx, device_value in enumerate(cpm_body_reordered["entry"]):
+        assert (
+            device_value["resource"]["resourceType"]
+            == ldap_body_reordered["entry"][idx]["resource"]["resourceType"]
+        ), f"ResourceTypes do not match. When calling with {item}"
+        for extension in device_value["resource"]["extension"]:
+            assert normalize_case(extension) in normalize_case(
+                ldap_body_reordered["entry"][idx]["resource"]["extension"]
+            ), f"Extension not in LDAP. When calling with {item}"
+        for identifier in device_value["resource"]["identifier"]:
+            assert normalize_case(identifier) in normalize_case(
+                ldap_body_reordered["entry"][idx]["resource"]["identifier"]
+            ), f"Identifier not in LDAP. When calling with {item}"
+        if item["path"] != "/Endpoint":
+            assert (
+                "owner" in device_value["resource"]
+            ), f"Key 'owner' does not exist in the device. When calling with {item}"
+            assert (
+                device_value["resource"]["owner"]
+                == ldap_body_reordered["entry"][idx]["resource"]["owner"]
+            ), f"Owners do not match. When calling with {item}"
+        else:
+            assert (
+                device_value["resource"]["managingOrganization"]
+                == ldap_body_reordered["entry"][idx]["resource"]["managingOrganization"]
+            )
+            assert (
+                device_value["resource"]["address"]
+                == ldap_body_reordered["entry"][idx]["resource"]["address"]
+            )
+
+
+@pytest.mark.parametrize(
+    "item",
+    _generate_test_data(
+        "sds_fhir_api.unique_queries.json",
+        test_count=int(os.getenv("TEST_COUNT", "10")),
+    ),
+)
+@pytest.mark.matrix
+def test_api_responses_match(item):
```

Comment:
is it possible to split this test into two tests - one for prod, one non-prod? my brain's melting following the flow - for example the line with 

```cpm_status, cpm_body = _request_dev(item)```

why doesn't that need `use_cpm` setting?

I'd maybe also consider splitting the test into 200 and non-200s if that's possible but not sure if that's complicated with the input data!

Reply:
There is another test for non-200s already. But I still have to catch them here as the unique_queries file doesn't always produce 200s.

dev env var USE_CPM=1 therefore filter for use_cpm is not needed.

eventually we won't need any tests for accessing dev. I would suggest when that time comes we remove that condition.